javalanglib/src/main/scala/java/lang/Class.scala

```diff
@@ -84,11 +85,7 @@ final class Class[A] private (data0: Object) extends Object {
 
   @inline // optimize for the Unchecked case, where this becomes identity()
   def cast(obj: Object): A = {
-    SemanticsUtils.asInstanceOfCheck({ () =>
-      (obj != null && !isJSType && !isInstance(obj))
-    }, { () =>
-      new ClassCastException("" + obj + " is not an instance of " + getName)
-    })
+    data.checkCast(obj)
```

Comment:
This is correct but it's less efficient than it used to be for the Unchecked case. Before, the optimizer was able to completely get rid of all the code, and the entire call to `cast(x)` collapsed to `x`. Now there is a call to `checkCast` (which itself doesn't do anything, but still).

That's not a deal-breaker, but we should at least remove the comment next to `@inline`, because it's clearly wrong now.

We could recover the previous optimization by making `Class.cast` an intrinsic; I'm not sure whether it's worth it, though.